foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/policy/PropertyRegenerationPolicyManager.java

```diff
@@ -0,0 +1,190 @@
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Class used for loading and configuring policies for determining when to regenerate a given {@link Property}.
+ */
+public class PropertyRegenerationPolicyManager extends AbstractPolicyManager {
+
+    private static final String TARGETS_REGEX = "/([^/]+)/([^/]+)";
+
+    private Set<PropertyRegenerationPolicy> propertyRegenerationPolicies;
+
+    /**
+     * Gets the property regeneration specific policies.
+     * @return map of policies
+     */
+    public Set<PropertyRegenerationPolicy> getPropertyRegenerationPolicies() {
+        if(this.propertyRegenerationPolicies == null) {
+            this.propertyRegenerationPolicies = new HashSet<>();
+            Map<String, Policy> policies = super.getPolicies();
+            if(policies != null) {
+                for (Map.Entry<String, Policy> entry : policies.entrySet()) {
+                    Policy value = entry.getValue();
+                    if(value instanceof PropertyRegenerationPolicy) {
+                        this.propertyRegenerationPolicies.add((PropertyRegenerationPolicy) value);
+                    }
+                }
+            }
+        }
+
+        return this.propertyRegenerationPolicies;
+    }
+
+    /**
+     * Method that allows subclasses to override the type reference with a
+     * subclass.
+     * 
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public Class getDeserializationClass() {
+        return PropertyRegenerationPolicyInput.class;
+    }
+
+    /**
+     * Method that allows subclasses to use an extended type for the policy
+     * class.
+     * 
+     * @param identifier
+     * @return
+     */
+    @Override
+    public DefaultPolicy createPolicy(String policyIdentifier) {
+        return new DefaultPropertyRegenerationPolicy(policyIdentifier);
+    }
+
+    /**
+     * Method that allows subclasses to set any additional configurations while
+     * reading a policy.
+     */
+    @Override
+    public void setAdditionalConfigurations(Policy policy, PolicyInput input) {
+        
+        // Make sure that the policy is the type we're expecting
+        if (policy instanceof DefaultPropertyRegenerationPolicy && input instanceof PropertyRegenerationPolicyInput) {
+            
+            // Set the additional configurations
+            DefaultPropertyRegenerationPolicy regenerationPolicy = (DefaultPropertyRegenerationPolicy) policy;
+            PropertyRegenerationPolicyInput regenerationPolicyInput = (PropertyRegenerationPolicyInput) input;
+
+            regenerationPolicy.setTargetPropertyKeys(convertTargetsToPropertyKeys(regenerationPolicyInput.getAnyTargets()));
+            regenerationPolicy.setPropertyRegenerationRules(convertInputRulesToPropertyRegenerationRules(regenerationPolicyInput.getRules()));
+            regenerationPolicy.setRegenerationStrategy(convertInputRuleToPropertyRegenerationStrategy(regenerationPolicyInput.getRegenerationMethod()));
```

Comment:
```suggestion
            regenerationPolicy.setPropertyRegenerationRules(convertInputRulesToPropertyRegenerationRules(regenerationPolicyInput.getRules()));
            regenerationPolicy.setRegenerationStrategy(convertInputRegenMethodToPropertyRegenerationStrategy(regenerationPolicyInput.getRegenerationMethod()));
```

A: would this refactoring make it a bit more consistent?

Reply:
Herm, I see, but the regen method is still a `PolicyInputRule` type so maybe not